nomos-http-api/src/http/backend/axum.rs

```diff
@@ -0,0 +1,103 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use full_replication::Blob;
+use hyper::StatusCode;
+use nomos_core::da::blob;
+use nomos_mempool::{openapi::Status, MempoolMetrics};
+use overwatch_rs::overwatch::handle::OverwatchHandle;
+
+use axum::{extract::State, response::IntoResponse, routing, Json, Router, Server};
+use utoipa::OpenApi;
+use utoipa_swagger_ui::SwaggerUi;
+
+use crate::Backend;
+
+use super::super::da;
+
+#[derive(Clone)]
+pub struct AxumBackendSettings {
+    pub da: OverwatchHandle,
+    pub addr: SocketAddr,
+}
+
+pub struct AxumBackend {
+    settings: Arc<AxumBackendSettings>,
+}
+
+#[derive(OpenApi)]
+#[openapi(
+    paths(
+        da_metrics,
+        da_status,
+    ),
+    components(
+        schemas(Status, MempoolMetrics)
+    ),
+    tags(
+        (name = "da", description = "data availibility related APIs")
+    )
+)]
+struct ApiDoc;
+
+type Store = Arc<AxumBackendSettings>;
+
+#[async_trait::async_trait]
+impl Backend for AxumBackend {
```

Comment:
this can go in a separate PR

Reply:
The merged PR has a test using Axum, but it is not in the normal code.

Reply:
Aaah, so yes, this axum implementation should go in a different pr.